utils/timer/eta.go

```diff
@@ -3,7 +3,20 @@
 
 package timer
 
-import "time"
+import (
+	"encoding/binary"
+	"time"
+)
+
+// ProgressFromHash returns the progress out of MaxUint64 assuming [b] is a key
+// in a uniformly distributed sequence that is being iterated lexicographically.
+func ProgressFromHash(b []byte) uint64 {
+	// binary.BigEndian.Uint64 will panic if the input length is less than 8, so
+	// pad 0s as needed.
```

Comment:
can we document this precondition in the comment above the function definition? That way it'll show up in IntelliSense.

Reply:
Ah - read this function too quickly 👍🏾